4주차/[Lv.3]_등굣길/주다현.js

```diff
@@ -0,0 +1,17 @@
+//런타임 에러가 나는 코드랍니다... 아직 해결하지 못했어요ㅠㅠㅠㅠ
+
+function solution(m, n, puddles) {
+  const arr = Array.from(Array(n), () => Array(m).fill(1));
+  puddles.forEach((v) => {
+    const [x, y] = v;
+    arr[x - 1][y - 1] = 0;
+  });
```

Comment:
```suggestion
  });
  
  for (let i = 0; i < m; i++) {
    if (arr[0][i] === null) {
      break;
    }
    
    arr[0][i] = 1;
  }
  
  for (let i = 0; i < n; i++) {
    if (arr[i][0] === null) {
      break;
    }
    
    arr[i][0] = 1;
  }
```
`n`, `m` 모두 각 `index`가 `0`일 경우 갈 수 있는 경로의 수는 모두 `1`입니다. 하지만 만약 `물에 잠긴 지역`이 존재한다면 그 지역 이후의 `index`에 해당하는 값은 0이 될 수 밖에 없습니다.